redwood-protocol-host/src/commonMain/kotlin/app/cash/redwood/protocol/widget/ProtocolBridge.kt

```diff
@@ -97,11 +97,11 @@ public class ProtocolBridge<W : Any>(
 
         is ModifierChange -> {
           val node = node(id)
-          val value = node.widget.value
 
           val modifier = change.elements.fold<_, Modifier>(Modifier) { outer, element ->
             val inner = factory.createModifier(element)
             if (inner is Modifier.UnscopedElement) {
+              val value = node.widget.value
```

Comment:
This fixes a crash in my test because WidgetValue’s `node.widget.value` isn’t available until all the properties are assigned.

Reply:
You'll get the same crash on anything using an unscoped modifier though. The fold is synchronous.

You mean the MutableXyz widgets which produce WidgetNodes as their value is the cause of the crash? We might have to change their view type to use Unit, then, and have a separate function to snapshot the values which walks the tree. We'll need to be able to also wrap this value in the future here, not just reference it.